src/checks/state-checks.ts

```diff
@@ -51,18 +52,36 @@ export const stateChecks = (
     return errorMessages;
 }
 
-const getPropertyCount = ({props, state}: {
+const getPropertyCount = ({props, object}: {
     props: string[],
-    state: State,
-}) : number => {
-    return props.map((prop) => prop in state ? 1 : 0)
+    object: Record<string, unknown>,
+}): number => {
+    return props.map((prop) => prop in object ? 1 : 0)
         .map((val) => Number(val))
         .reduce((prev, curr) => prev + curr, 0)
 }
 
-export const AtMostOne = ({props, errorCode}:{props: string[], errorCode: StateMachineErrorCode}): StateChecker => {
+export const AtMostOne = ({props, errorCode, path}: {
+    props: string[],
+    errorCode: StateMachineErrorCode,
+    // path to a sub-property within the state to use as the
+    // context for the property checks. This is intended to
+    // support enforcement of constraints on nested properties
+    // within a State.
+    // See the Map's ItemReader.ReaderConfiguration at most one
+    // rule for MaxItems and MaxItemsPath.
+    path?: string
+}): StateChecker => {
```

Comment:
using a path param here to get the nested props keeps the rules in the `validator.ts` file a bit simpler